pkg/cluster/kubeconfig.go

```diff
@@ -115,47 +117,58 @@ func (m *manager) generateKubeconfigs(ctx context.Context) error {
 	}
 
 	m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
-		doc.OpenShiftCluster.Properties.AdminKubeconfig = adminInternalClient.File.Data
-		doc.OpenShiftCluster.Properties.AROServiceKubeconfig = aroServiceInternalClient.File.Data
-		doc.OpenShiftCluster.Properties.AROSREKubeconfig = aroSREInternalClient.File.Data
-		doc.OpenShiftCluster.Properties.UserAdminKubeconfig = aroUserInternalClient.File.Data
+		doc.OpenShiftCluster.Properties.AdminKubeconfig = adminInternalClient
+		doc.OpenShiftCluster.Properties.AROServiceKubeconfig = aroServiceInternalClient
+		doc.OpenShiftCluster.Properties.AROSREKubeconfig = aroSREInternalClient
+		doc.OpenShiftCluster.Properties.UserAdminKubeconfig = aroUserInternalClient
 		return nil
 	})
 	return err
 }
 
-func generateKubeconfig(pg graph.PersistedGraph, commonName string, organization []string, validity time.Duration, internal bool) (*kubeconfig.AdminInternalClient, error) {
-	var ca *tls.AdminKubeConfigSignerCertKey
-	var adminInternalClient *kubeconfig.AdminInternalClient
-	err := pg.Get(true, &ca, &adminInternalClient)
+func generateKubeconfig(pg graph.PersistedGraph, commonName string, organization []string, validity time.Duration, internal bool) ([]byte, error) {
+	var ca *installer.AdminKubeConfigSignerCertKey
+	var adminInternalClient *installer.AdminInternalClient
+	err := pg.GetByName(false, "*tls.AdminKubeConfigSignerCertKey", &ca)
```

Comment:
are we using a pointer of a pointer here? It seems a little weird...

Reply:
the type name as-saved in the map is "*tls.AdminKubeConfigSignerCertKey" and then we pull it out to a struct via a pointer, yeah